iceoryx_binding_c/source/c_publisher.cpp

```diff
@@ -29,15 +29,17 @@ extern "C" {
 }
 
 iox_pub_t iox_pub_init(iox_pub_storage_t* self,
-                       const char* service,
-                       const char* instance,
-                       const char* event,
-                       const uint64_t historyCapacity)
+                       const char* const service,
+                       const char* const instance,
+                       const char* const event,
+                       const uint64_t historyCapacity,
+                       const char* const nodeName)
 {
     new (self) cpp2c_Publisher();
     iox_pub_t me = reinterpret_cast<iox_pub_t>(self);
-    iox::popo::PublisherOptions options;
+    PublisherOptions options;
     options.historyCapacity = historyCapacity;
+    options.nodeName = NodeName_t(TruncateToCapacity, nodeName);
```

Comment:
I decided against introducing a pub/sub options struct similar to the PublisherOptions in the C++ API because in C we cannot ensure that the struct is initialized and therefore introduce undefined behavior when assigning to the C++ PublisherOptions.

Reply:
While it's true that this can lead to undefined behavior, there is the same problem without using a struct
```c
const uint64_t historyCapacity;
const char nodeName[100];
iox_pub_storage_t publisherStorage;
iox_pub_t publisher = iox_pub_init(&publisherStorage, "Radar", "FrontLeft", "Object", historyCapacity, nodeName)
```

The C people have to deal with this stuff all the time and it's not uncommon to call an init function before a struct can be used. So the example above would look like this with a struct
```c
iox_pub_options_t publisherOptions;
iox_pub_init_options(publisherOptions);
publisherOptions.historyCapacity  = 10U;
strncpy(publisherOptions.nodeName, "iox-c-publisher-node");
iox_pub_t publisher = iox_pub_init(&publisherStorage, "Radar", "FrontLeft", "Object", &publisherOptions);
```
Using default options if a nullptr was passed to `iox_pub_init` instead of `publisherOptions`;
I think this can be done with the API cleanup issue, though.

Reply:
For parameter only you would basically have to set them all because not all of them are null-able. With the struct you could decide that passing a `nullptr` is equal to a struct where only `init` was called.
`init` would also set the same defaults like in the C++ API

Reply:
@elBoberido you've convinced me. This will be done in #408.